### PR TITLE
Patch 1/support rewriting rpc url wlt 2739

### PR DESCRIPTION
**Pull request by everdimension**

### CODE REVIEW

File: src/ui/pages/Networks/NetworkUpdateSuccess/NetworkUpdateSuccess.tsx [hunk 1]
```diff
@@ -10,19 +10,25 @@ import { noValueDash } from 'src/ui/shared/typography';
...
 }: {
   network: NetworkConfig;
-  oldNetwork: NetworkConfig;
+  prevNetwork: NetworkConfig;
   onClose: () => void;
 }) {
+  const { networks } = useNetworks();
+
+  const prevRpcUrl = Networks.getNetworkRpcUrlInternal(prevNetwork);
+  const rpcUrl = networks?.getRpcUrlInternal(createChain(network.chain));
```
Comment (zerts) [inline src/ui/pages/Networks/NetworkUpdateSuccess/NetworkUpdateSuccess.tsx:30]: Looks like having getRpcUrlInternal for network as a param is also nice, cause now we are converting network into chain just to convert it back inside the function

Comment (zerts) [reply src/ui/pages/Networks/NetworkUpdateSuccess/NetworkUpdateSuccess.tsx:30]: Or may be we can just do `Networks.getNetworkRpcUrlInternal(network);` here?

File: src/ui/pages/AddEthereumChain/RpcUrlForm/RpcUrlForm.tsx [hunk 2]
```diff
@@ -46,24 +51,34 @@ const Field = React.forwardRef(
...
-  oldNetwork: NetworkConfig;
+  prevNetwork: NetworkConfig;
   rpcUrlHelpHref: string;
   isSubmitting: boolean;
   onSubmit: (result: NetworkConfig) => void;
   onCancel: () => void;
 }) {
+  const { networks } = useNetworks();
+
+  const currentRpcUrl = networks?.getRpcUrlInternal(
```
Comment (zerts) [inline src/ui/pages/AddEthereumChain/RpcUrlForm/RpcUrlForm.tsx:69]: `Networks.getNetworkRpcUrlInternal(prevNetwork);`?

File: src/ui/pages/AddEthereumChain/RpcUrlForm/RpcUrlForm.tsx [hunk 3]
```diff
@@ -72,38 +87,51 @@ export function RpcUrlForm({
...
+          /**
+           * If network HAS `rpc_url_internal`, we introduce a `rpc_url_user` field
+           * as a mechanism to overwrite it,
+           * else - we use `rpc_url_public`
+           */
+          <Field
+            label="New RPC URL"
+            name="rpc_url_user"
+            type="url"
+            defaultValue={network.rpc_url_public?.[0] || ''}
```
[the remark this answers is not in the document]
Comment (zerts) [reply src/ui/pages/AddEthereumChain/RpcUrlForm/RpcUrlForm.tsx:123]: If the code is correct, then we could set only Field name with condition, without copying the whole component

[the remark this answers is not in the document]
Comment (zerts) [reply src/ui/pages/AddEthereumChain/RpcUrlForm/RpcUrlForm.tsx:123]: `<Field
            label="New RPC URL"
            name={prevNetwork.rpc_url_internal || prevNetwork.rpc_url_user ? "rpc_url_user" : "rpc_url_public[]"}
            type="url"
            defaultValue={network.rpc_url_public?.[0] || ''}
            required={true}
          />`

[the remark this answers is not in the document]
Comment (everdimension) [reply src/ui/pages/AddEthereumChain/RpcUrlForm/RpcUrlForm.tsx:123]: The `network` object is coming from a dapp (converted from `AddEthereumChainParameter`, therefore it only has `rpc_user_public` filled out. I agree though that it looks confusing here.

---

Having ternary with two fields in my experience feels more cohesive. When several props are changed based on a condition, it's helpful to see those props "together" for one case and then others "together" for another case. I hope this explanation makes sense :)

[the remark this answers is not in the document]
Comment (everdimension) [reply src/ui/pages/AddEthereumChain/RpcUrlForm/RpcUrlForm.tsx:123]: Ok you're right, it's just one prop here :)

**Review by zerts — APPROVED** [review]

File: src/ui/pages/AddEthereumChain/RpcUrlForm/RpcUrlForm.tsx (hunk 2 above)
Comment (vyorkin) [reply src/ui/pages/AddEthereumChain/RpcUrlForm/RpcUrlForm.tsx:69]: As I can see `getRpcUrlInternal` has an extra check for existing network

**Review by vyorkin — APPROVED** [review]

**Pull request merged**